docs/modules/deployment/pages/time-series-storage/timeseries/cortex.adoc

```diff
@@ -0,0 +1,91 @@
+
+= Cortex Time Series Plugin
+
+The Time Series Storage Plugin for Cortex helps {page-component-title} to persist aggregated flow data to Cortex.
+The plugin uses the xref:deployment:time-series-storage/timeseries/ts-integration-layer.adoc[time series storage integration layer] to convert metrics to a Prometheus model and delegate writes and reads to https://cortexmetrics.io/[Cortex].
+The plugin also makes it possible for https://docs.opennms.com/helm/latest/index.html[OpenNMS Helm] to visualize flow graphs from the Cortex instance.
+
+.Time Series Storage Plugin for Cortex architecture
+image::time-series-storage/cortex.png[Cortex plugin architecture,500]
+
+== Requirements
+
+Make sure you have the following before you start to use the Cortex plugin:
+
+* Horizon 30.0.0+ (Meridian availability with the 2023 release)
+* Cortex version 1.10.0+
+
+NOTE: We plan to keep the plugin compatible with https://grafana.com/oss/mimir/[Mimir], Grafana's fork of Cortex, as long as Mimir remains compatible with Cortex.
+The minimum supported Mimir version is 2.0.0.
+
+== Deployment
+
+To use the Time Series Storage Plugin for Cortex, do the following:
+
+.Start Cortex
+See https://cortexmetrics.io/docs/getting-started/.
+
+You can also download a https://github.com/opennms-forge/stack-play/tree/master/standalone-cortex-minimal[minimal, standalone version of Cortex] and start Cortex with `docker-compose up`.
+
+.Build and install the plugin into your local Maven repository
+
+[source, console]
+----
+mvn clean install
+----
+
+.Enable and configure the time series storage
+
+[source, console]
+----
+echo "org.opennms.timeseries.strategy=integration
+org.opennms.timeseries.tin.metatags.tag.node=${node:label}
+org.opennms.timeseries.tin.metatags.tag.location=${node:location}
+org.opennms.timeseries.tin.metatags.tag.geohash=${node:geohash}
+org.opennms.timeseries.tin.metatags.tag.ifDescr=${interface:if-description}
+org.opennms.timeseries.tin.metatags.tag.label=${resource:label}" >> ${OPENNMS_HOME}/etc/opennms.properties.d/cortex.properties
+----
+
+.Install the plugin from the OpenNMS Karaf shell
+
+[source, console]
+----
+feature:repo-add mvn:org.opennms.plugins.timeseries/cortex-karaf-features/1.0.0-SNAPSHOT/xml
+feature:install opennms-plugins-cortex-tss
+----
+
+.Configure
+Note that you can omit this step if you use the default values.
+
+[source, console]
+----
+config:edit org.opennms.plugins.tss.cortex
+
+property-set writeUrl http://localhost:9009/api/prom/push
+property-set readUrl http://localhost:9009/prometheus/api/v1
+property-set maxConcurrentHttpConnections 100
+property-set writeTimeoutInMs 1000
+property-set readTimeoutInMs 1000
+property-set metricCacheSize 1000
+property-set externalTagsCacheSize 1000
+property-set bulkheadMaxWaitDurationInMs 9223372036854775807
+
+config:update
+----
+
+.Update automatically
```

Comment:
This step is specific to developer environments, and should also go away once we're building and releasing the plugin.